docs/v1/P4Runtime-Spec.mdk

```diff
@@ -4496,6 +4496,51 @@ multi-reader lock (also known as multiple-readers/single-writer lock).
 Conversely (&eg; in a single-threaded architecture), it may choose to serialize
 `Read` RPC processing.
 
+## Parallelism of read and write requests
+
+A P4Runtime server may be implemented to serve at most one
+`ReadRequest` or `WriteRequest` message at a time, sequentially.  It
+may also serve multiple requests in parallel, and it is expected that
+some client software would be easier to implement with good
+performance characteristics if a server did so.
+
+For example, imagine a client that wanted to use `WriteRequest`
+messages with large batches of insert, modify, and/or delete
+operations on an IP route table, in order to achieve higher throughput
+of updates to this table.  Such a client might also wish to send
+`WriteRequest` messages with small numbers of updates to an
+`ActionSelector` object that controlled which links were in which
+LAGs, and have those small requests start processing even if there is
+a large `WriteRequest` batch currently being processed, but it will
+not complete for a significant amount of time.
+
+The restrictions on which a client may rely are:
+
+* The processing of any two `WriteRequest` messages `W1` and `W2` must
+  result in the same state as if one of them was completed before the
+  other began.
+* For any `WriteRequest` `W` and any `ReadRequest` `R`, `R` must
+  return results consistent with a state where `W` has completed
+  processing, or `W` has not yet begun processing.
+
+For example, if a P4Runtime server maintained, independently for each
+device it managed, a separate multi-reader single-writer lock for each
+stateful object in the P4 program, and before starting the processing
+of a `WriteRequest` message it acquired a write lock for each stateful
+object affected by the `WriteRequest`, and before starting the
+processing of a `ReadRequest` message it acquired a read lock for each
+stateful object accessed by the `ReadRequest`, such an implementation
+should meet all of the requirements above.
```

Comment:
Why does it say "should meet" instead of "meets"?  Is there anything else to be considered?

Reply:
The only reason it says that is my often-cautious way of writing when I'm not yet sure of something.  Changing it to "meets" sounds like a good idea.  We can correct it later if I messed it up :)

Reply:
No worries, thank for updating and writing all of this in the first place.